### DISCUSSION TITLE
Cannot switch view in page

### DISCUSSION THREAD

**Discussion by Lan-Phan** [illacloud/illa-builder#1024] [General]
Hello,

I create 1 sample page which body contains 2 views: view 1 and view 2
+ view 1 has text "Hello view 1"
+ view 2 has text "Hello view 2"

Then I deploy the app:
+ Select view 1, default view "View 1" -> deploy and see "Hello view 1"
+ Select view 2, default view "View 1" -> deploy and see "Hello view 1"
+ Select view 1, default view "View 2" -> deploy and see "Hello view 1"
+ Select view 2, default view "View 2" -> deploy and see "Hello view 1"

My question is, how can I deploy and see view 2 "Hello view 2"? What's main purpose of this view concept? Is it possible to use view for responsive design (view 1 for smart watch, view 2 for mobile, view 3 for desktop ...)?

**Comment by AruSeito** [comment]
Thanks for your feedback,I think this is bug,we will fix it soon.
At that time, we envisioned using it to display different pages according to different roles and permissions
But later we found that this usage is too complex, even we will be confused, so we will adjust this in the near future

**Comment by Lan-Phan** [comment]
Hi @AruSeito ,

How's about my suggestion to switch view based on size of display? I think it's possible and it provides perfect responsive design for user

**Reply by wenzhuo2022** [reply]
Hello @Lan-Phan,  
Thanks for your suggestion. That's a good idea. We will try to support different screen sizes. The current naming of the "view" feature is indeed ambiguous. In fact, "view" doesn't work for different screen sizes. 

Our documentation page is a case about how the view works: https://www.illacloud.com/en-US/docs/about-illa. You can click the menu to change views without loading page resources on this page. 
You can also check our doc about the "Page" feature where we explain how pages and views work: https://www.illacloud.com/en-US/docs/page

**Comment by AruSeito** [comment]
Hello @Lan-Phan ,
Your suggestions have been conveyed to my colleague. For details, you can communicate with my colleague @wenzhuo2022   .
